app/pages/settings/email.jsx

```diff
@@ -75,14 +75,17 @@ function ProjectPreferences({ projects, projectPreferences, onChange }) {
             <tr key={projectPreference.id}>
               <td>
                 <input
+                  id={projectPreference.id}
                   type="checkbox"
                   name="email_communication"
                   checked={projectPreference.email_communication}
                   onChange={e => onChange(i, e)}
                 />
               </td>
               <td>
-                {projectNames[projectPreference.links.project]}
+                <label htmlFor={projectPreference.id}>
+                  {projects[i].display_name}
```

Comment:
This isn’t the project name that was being displayed by the deleted line.

Reply:
But the deleted line doesn't seem to work either, since the bug can be reproduced on the live site.

Reply:
That bug is probably because the index here refers to position in a filtered copy of the preferences array, but the change handler acts on the unfiltered array.

Reply:
Either way, the project name is wrong here.